src/FlagProvider.tsx

```diff
@@ -14,50 +14,49 @@ const FlagProvider: React.FC<React.PropsWithChildren<IFlagProvider>> = ({
   config,
   children,
   unleashClient,
-  startClient = true,
+  startClient = typeof window !== undefined,
 }) => {
-  const client = React.useRef<UnleashClient>(unleashClient);
-  const [flagsReady, setFlagsReady] = React.useState(false);
-  const [flagsError, setFlagsError] = React.useState(null);
-  const flagsErrorRef = React.useRef(null);
-  const callbackRegisteredRef = React.useRef(null);
-
   if (!config && !unleashClient) {
     console.warn(
-      `You must provide either a config or an unleash client to the flag provider. If you are initializing the client in useEffect, you can avoid this warning by
-      checking if the client exists before rendering.`
+      `You must provide either a config or an unleash client to the flag provider. 
+      If you are initializing the client in useEffect, you can avoid this warning 
+      by checking if the client exists before rendering.`
     );
   }
 
-  if (!client.current) {
-    client.current = new UnleashClient(config);
-  }
+  const client = React.useRef<UnleashClient>(
+    unleashClient || new UnleashClient(config as IConfig)
+  );
+  const [flagsReady, setFlagsReady] = React.useState(false);
+  const [flagsError, setFlagsError] = React.useState(null);
+  const isStartedRef = React.useRef(false); // prevent double instantiation, https://github.com/reactwg/react-18/discussions/18
```

Comment:
I don't understand why we need this added complexity. The behavior of strict mode is intentionally rendering twice in order to help you catch bugs in useEffects. Our useEffects should currently be handling these cases, because we defensively stop the client before starting it again, so if it renders twice in development (as intended by React) it should behave as a single instance. 

This pattern is documented here: https://beta.reactjs.org/learn/synchronizing-with-effects#how-to-handle-the-effect-firing-twice-in-development

If there are any useEffects misbehaving in development, the correct way to handle it in my mind would be to architect them so that running them twice is a noop. Either by implementing defensive logic, or implementing a cleanup function.